node/src/components/consensus/protocols/zug.rs

```diff
@@ -2599,3 +2620,65 @@ mod specimen_support {
         }
     }
 }
+
+mod registered_sync {
+    use crate::{
+        types::{DataSize, NodeRng},
+        utils::specimen::{Cache, LargestSpecimen, SizeEstimator},
+    };
+    use casper_types::{TimeDiff, Timestamp};
+    use rand::Rng as _;
+    use serde::{Deserialize, Serialize};
+    use std::collections::BTreeMap;
+
+    #[derive(Default, DataSize, Clone, Serialize, Deserialize, Debug, PartialEq, Eq, Hash)]
+    pub struct RegisteredSync(BTreeMap<RandomId, Timestamp>);
+
+    #[derive(
+        DataSize, Copy, Clone, Serialize, Deserialize, Debug, PartialEq, Eq, Hash, Ord, PartialOrd,
+    )]
+    pub struct RandomId(u64);
+
+    impl RegisteredSync {
+        /// Prunes entries older than one minute.
+        fn prune_old(&mut self) {
+            const ONE_MIN: TimeDiff = TimeDiff::from_seconds(60);
+
+            self.0.retain(|_, timestamp| timestamp.elapsed() < ONE_MIN);
+        }
+
+        pub fn create_and_register_new_id(&mut self, rng: &mut NodeRng) -> RandomId {
+            self.prune_old();
+
+            let id = loop {
+                let id = RandomId::new(rng);
+
+                if self.0.contains_key(&id) == false {
+                    break id;
+                }
+            };
+
+            self.0.insert(id, Timestamp::now());
+
+            id
+        }
+
+        /// Returns `true` if ID has been succesfully removed, *i.e.* it was present
+        /// in the data structure.
+        pub fn try_remove_id(&mut self, id: RandomId) -> bool {
+            self.0.remove(&id).is_none()
```

Comment:
I think the comment is reversed, isn't it? That is, this seems to return `true` when `remove()` returned `None`, which is when the ID _wasn't_ present.
I actually think it would make more sense the way the comment is written, ie. I'd change `is_none()` to `is_some()` and leave the comment as is.

Reply:
Done